compare.py

```diff
@@ -0,0 +1,413 @@
+        'customfield_11833': url
+    }
+
+    try:
+        new_issue = jira_client.create_issue(fields=issue_dict)
+    except JIRAError as e:
+        logging.error("Could not create Jira ticket [{}]".format(issue_dict))
+        logging.error("{}".format(e))
+
+    for attachment in attachments:
+        if os.path.exists(attachment):
+            try:
+                jira_client.add_attachment(issue=new_issue, attachment=attachment)
+            except JIRAError as e:
+                logging.error("Could not add attachment [{}]".format(attachment))
+                logging.error("{}".format(e))
+        else:
+            logging.error("Could find file to attach to Jira Ticket: [{}]".format(attachment))
+
+    logging.info("\tCreated jira ticket: [{}]".format(new_issue.key))
+
+    return
+
+
+def append_contours_to_masks(contours, mask_path):
+
+    with open(str(mask_path, 'rb')) as pkl:
+        diff_contours = pickle.load(pkl)
+
+    diff_contours.append(contours)
+
+    with open("{}/diff_contours.pkl".format(mask_path), 'wb') as pkl:
+        pickle.dump(diff_contours, pkl)
+
+    return 1
+
+
+def process_alerts(root,
+                   json_alerts="website_changes_log.json", back_date=date.today(), skip=0):
+
+    logging.info("Processing root folder: {}".format(root))
+
+    count = 1
+    jira_count = 0
+
+    website_changes_log = open("{}{}{}".format(root, os.sep, json_alerts), "r")
+
+    for change in website_changes_log:
+
+        change = json.loads(change)
+        url = change["url"]
+        folder = convert_url_to_folder(url)
+
+        scan_date = change["Date"]
+        scan_date = datetime.strptime(scan_date, '%Y%m%d-%H%M')
+
+        if scan_date.date() < back_date:
+            continue
+
+        logging.info("[{}] Processing folder: {}".format(count, "{}{}{}".format(root, os.sep, folder)))
+
+        datum_image = None
+        latest_image = None
+
+        if os.path.isdir("{}{}{}".format(root, os.sep, folder)) and count >= skip:
+            for file in os.listdir("{}{}{}".format(root, os.sep, folder)):
+                if file == 'imageB.png':
+                    # imageB is the datum
+                    datum_image = file
+                elif file == 'imageA.png':
+                    # imageA is the latest image
+                    latest_image = file
+
+            # todo: remove
+            if debug:
+                datum_image = "imageB.png"
+                latest_image = "imageB.png"
+
+            if datum_image is not None and latest_image is not None:
+                jira_ticket_details = compare(folder="{}{}{}".format(root, os.sep, folder))
+
+                if jira_ticket_details:
+                    jira_count = jira_count + 1
+
+        count = count + 1
+
+    logging.info("Processed a total of [{}] changes and WOULD HAVE created [{}] jira tickets".format(count, jira_count))
+
+
+def convert_url_to_folder(url):
+    folder = url.replace(".", "")
```

Comment:
all replace could be done in single line using regex `re.sub('[https:,?,/,.]', '', url)`